### DISCUSSION TITLE
Device_id for force_device_availability

### DISCUSSION THREAD

**Discussion by WallyDW** [SukramJ/aiohomematic#865] [Q&A]
If you want to use the force_device_availability service call in HA the call then request a device_id.
I can find it in the HA GUI by manually selecting a device, but I have found no way to find the device_id programatically.
It seems kind of wrong that I might have to click all the devices in the developer tools, write down the device_ids and then hardcode them into my scripts.

**Comment by SukramJ** [comment]
The `device_id` is the preferred way in HA automations to make a device _selectable in the UI_.

> and then hardcode them into my scripts.

Why don't you use the blueprints?

**Comment by WallyDW** [comment]
Because sometimes a device is not listed as unreachable, but the timestamp on the UNREACH paramset is then not updated.

**Reply by SukramJ** [reply]
The latest version supports an alternative device address

**Comment by WallyDW** [comment]
I am not sure how it works, since there is no explanation to what Device Address actually is.
Is it just the serial number of the device, which is also the address used in the XMLAPI output? The length of the number in the example does not fit the length of the serial number listed in HomeMatic/XMLAPI.

**Reply by SukramJ** [reply]
XMLAPI is irrelevant and not used.
Every entity has an attribute named `address`. The first part, before the : is the device_address.
in the Homematic WebUI it is called serial number.

That the reason why the device_id is mainly used: It can be **selected in the UI** without copying text.

**Reply by WallyDW** [reply]
How do I mark this reply as an answer?

**Comment by WallyDW** [comment]
Perfect!
That is also the serial number of the device and thereby the address listed in XMLAPI.
I have therefore several places to extract this information.
Thanks a lot!
